crates/turborepo-lib/src/rewrite_json.rs

```diff
@@ -0,0 +1,420 @@
+                }
+                None => (current_path, current_node),
+            }
+        }
+        // All other node types are complete.
+        _ => (current_path, current_node),
+    }
+}
+
+/**
+ * Given path segments, generate a JSON object.
+ */
+fn generate_object(path_segments: &[&str], value: &str) -> String {
+    let mut output = String::new();
+    let length = path_segments.len();
+
+    for path in path_segments {
+        output.push('{');
+        output.push('\"');
+        output.push_str(path);
+        output.push('\"');
+        output.push(':');
+    }
+    output.push_str(value);
+    for _ in 0..length {
+        output.push('}');
+    }
+
+    output
+}
+
+/**
+ * Given path segments, generate a JSON object member with an optional
+ * trailing separator.
+ */
+fn generate_member(path_segments: &[&str], value: &str, separator: &str) -> String {
+    let (key, remainder) = path_segments.split_first().unwrap();
+    let object = generate_object(remainder, value);
+    format!("\"{key}\":{object}{separator}")
+}
+
+/**
+ * Given a JSONC document and an object traversal path, `unset_path` will
+ * return a minimally-mutated JSONC document with all occurrences of the
+ * specified path removed.
+ */
+pub fn unset_path(
+    json_document_string: &str,
+    path: &[&str],
+) -> Result<(bool, String), RewriteError> {
+    let root = get_root(json_document_string)?;
+
+    // The key path can appear multiple times. This a vec that contains each time it
+    // occurs.
+    let path_ranges = find_all_paths(&root, &path, &mut vec![]);
+
+    if path_ranges.is_empty() {
+        return Ok((false, json_document_string.to_owned()));
+    }
+
+    // We mutate this as we go.
+    let mut output: String = json_document_string.to_owned();
+
+    // We could either join overlapping ranges, or just carry it over.
+    // This elects to carry it over.
+    let mut last_start_position = None;
+
+    // We iterate in reverse since we're mutating the string.
+    path_ranges.iter().rev().for_each(|range| {
+        let end = match last_start_position {
+            Some(last_start_position) => {
+                if range.end > last_start_position {
+                    last_start_position
+                } else {
+                    range.end
+                }
+            }
+            None => range.end,
+        };
+
+        let is_overlapping = end != range.end;
+        let replacement_char = if is_overlapping {
+            ""
+        } else {
+            &range.replacement_char
+        };
+
+        output.replace_range(range.start..end, replacement_char);
+        last_start_position = Some(range.start);
+    });
+
+    Ok((true, output))
+}
+
+/**
+ * find_all_paths returns the list of ranges which define the specified
+ * token.
+ */
+fn find_all_paths<'a>(
+    current_node: &'a jsonc_parser::ast::Value<'a>,
+    target_path: &[&'a str],
+    current_path: &mut Vec<&'a str>,
```

Comment:
Not sure if I'm missing something, but this argument seems to never be read? I ripped it out and all the tests continued to pass once I updated the callsites.

Reply:
Ahh, vestigial code from when I copy/pasted the method signature from `get_closest_node`. 😜

Reply:
(Fantastic catch.)